packages/block-library/src/navigation/edit/index.js

```diff
@@ -157,19 +153,19 @@ function Navigation( {
 		status: createNavigationMenuStatus,
 		error: createNavigationMenuError,
 		value: createNavigationMenuPost,
+		isPending: isCreatingNavigationMenu,
+		isSuccess: createNavigationMenuIsSuccess,
+		isError: createNavigationMenuIsError,
```

Comment:
```suggestion
		didSucceed: createNavigationMenuIsSuccess,
		didError: createNavigationMenuIsError,
```

Reply:
Amusingly this is _exactly_ what I did locally but then I decided it might be confusing. Happy to be told otherwise though! 😄

Reply:
Oh, It's not that I prefer the names, this is what `useCreateNavigationMenu` returns later on in your PR. I think you must have renamed in one place but not the other.

I think why the navigation block e2e tests are failing here.

Reply:
It would work, but I wonder if this could be simpler. `create` already deals with promise rejections internally, but then it converts them to these error flags that we then propagate to `useEffect`. Would it be possible to get rid of all the didSucceed/didError bools and do something like this?

```js
function onCreateMenu() {
	try {
		hideNavigationMenuCreateNotice();
		speak( __( `Creating Navigation Menu.` ) );

		await create();
	
		setRef( createNavigationMenuPost.id );
		selectBlock( clientId );

		showNavigationMenuCreateNotice(
			__( `Navigation Menu successfully created.` )
		);
	} catch( e ) {
		showNavigationMenuCreateNotice(
			__( 'Failed to create Navigation Menu.' )
		);
	}
}
```

It should take less lines to express and be easier to debug – the entire logic is in one place here.

Reply:
I think I did it this way because previously the UI was hanging because we were calling async functions as callbacks.

So we'd call `onCreateMenu` in a `onClick` and then the UI would hang with no feedback whilst the operation took place async.

With my method it was easier to reflect the async nature in the UI.

Reply:
> Oh, It's not that I prefer the names, this is what `useCreateNavigationMenu` returns later on in your PR. I think you must have renamed in one place but not the other.

A good spot. I've renamed the exported vars to use `is` as this is what I wanted. Perhaps I forgot to push a commit 🤔